### PR TITLE
implemented user registration , dynamic user list , edit , update and…

### PR DESCRIPTION
**Pull request by merlouduron**
… delete function with validations

# Pull Request Review Comments

## Overview
- **Purpose of the PR**:
    - [ ] Feature addition
    - [ ] Bug fix
    - [ ] Refactoring
    - [ ] Documentation update
    - [ ] Other (please specify below)

## Specific Areas for Verification
- Please specify any particular requests or questions for the reviewer here.

## Suggestions/Concerns
- If you have any suggestions or concerns, please mention them here.

---

_**Additional Comments for the Reviewer (Optional)**_

- Add any additional comments or information for the reviewer here.

### CODE REVIEW

File: client/src/pages/Users/UsersRegistration.tsx [hunk 1]
```diff
@@ -74,63 +76,86 @@ const UsersRegistration = () => {
...
-    //       }
-    //     }
-    //   }
+      const handleSubmit = async (e) => {
+        e.preventDefault()
+        const token = localStorage.getItem('accessToken')
+        const { username, password, confirm_password, email , confirm_email} = userData
+
+        if (!username || !email || !password || !confirm_password) {
+          alert('All fields are required.')
```
Comment (edward-stalley-work) [inline client/src/pages/Users/UsersRegistration.tsx:85]: I was able to register without a last name.
All Inputs must be filled in.

File: client/src/pages/Users/UsersRegistration.tsx [hunk 2]
```diff
@@ -74,63 +76,86 @@ const UsersRegistration = () => {
...
+            'Please enter your last name and first name using the English alphabet, with an underscore between them.',
+          )
+          return
+        } else {
+          setIsUsernameValid(true)
+        }
+
+        if (password !== confirm_password) {
+            setIsPasswordMatch(false) 
+            alert('Passwords do not match.')
```
Comment (edward-stalley-work) [inline client/src/pages/Users/UsersRegistration.tsx:102]: All the validation messages are in English.
Japanese mode needs to have Japanese messages.
I have updated the glossary with all the messages being used in Users.

File: client/src/pages/Users/UsersRegistration.tsx [hunk 3]
```diff
@@ -74,63 +76,86 @@ const UsersRegistration = () => {
...
+          setIsEmailValid(false)
+          alert('Please enter a valid email address.')
+          return
+        } else {
+          setIsEmailValid(true)
+        }
+
+        if (email !== confirm_email) {
+          setIsEmailMatch(false)
+          alert('Email provided do not match.')
```
Comment (edward-stalley-work) [inline client/src/pages/Users/UsersRegistration.tsx:130]: Add an 's'.
'Emails provided do not match.'

File: client/src/pages/Users/UsersRegistration.tsx [hunk 4]
```diff
@@ -12,20 +13,23 @@ const UsersRegistration = () => {
     const [activeTabOther, setActiveTabOther] = useState('users')
     const storedUserID = localStorage.getItem('userID')
     const { language, setLanguage } = useLanguage()
+    const [isUsernameValid, setIsUsernameValid] = useState(true)
+    const [isPasswordValid, setIsPasswordValid] = useState(true)
+    const [isPasswordMatch, setIsPasswordMatch] = useState(true)
+    const [isIsEmailMatch, setIsEmailMatch] = useState(true)
```
Comment (edward-stalley-work) [inline client/src/pages/Users/UsersRegistration.tsx:19]: Maybe no need for double 'isIs'.

'isEmailMatch'

File: client/src/pages/Users/UsersRegistration.tsx [hunk 5]
```diff
@@ -12,20 +13,23 @@ const UsersRegistration = () => {
...
+      {
+        username: '',
+        first_name: '',
+        last_name: '',
+        password: '',
+        email: '',
+        confirm_password: '',
+        confirm_email: '',
+      },
+    )
```
Comment (edward-stalley-work) [inline client/src/pages/Users/UsersRegistration.tsx:32]: Can we add date_joined automatically using the current DateTime?
Currently it is blank when a user is registered.

**Review by edward-stalley-work — CHANGES_REQUESTED** [review]
Looks Good Merlou thank you!
Just a few changes so please see comments.

File: client/src/pages/Users/UsersRegistration.tsx (hunk 1 above)
Comment (merlouduron) [reply client/src/pages/Users/UsersRegistration.tsx]: Fix.

File: client/src/pages/Users/UsersRegistration.tsx (hunk 2 above)
Comment (merlouduron) [reply client/src/pages/Users/UsersRegistration.tsx]: Fixed.

File: client/src/pages/Users/UsersRegistration.tsx (hunk 3 above)
Comment (merlouduron) [reply client/src/pages/Users/UsersRegistration.tsx]: Fixed.

File: client/src/pages/Users/UsersRegistration.tsx (hunk 4 above)
Comment (merlouduron) [reply client/src/pages/Users/UsersRegistration.tsx]: Fixed.

File: client/src/pages/Users/UsersRegistration.tsx (hunk 5 above)
Comment (merlouduron) [reply client/src/pages/Users/UsersRegistration.tsx:32]: This is already implemented. You can see date_joined in User list. 🙇

**Pull request merged**